core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/DetailedSampleBuilder.java

```diff
@@ -1151,4 +1151,19 @@ public boolean isSaved() {
     return false;
   }
 
+  @Override
+  public List<FileAttachment> getAttachments() {
+    throw new UnsupportedOperationException("Cannot attach files to non-persisted builder object");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Cannot attach files to non-persisted builder object" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)